### PR TITLE
Add support to disable tracing for HTTP requests

### PR DESCRIPTION
**Pull request by srnagar**
This PR adds support to disable tracing for specific HTTP requests. This is similar to https://github.com/Azure/azure-sdk-for-java/issues/14584 but allows control at the request level rather than at the environment level.

### CODE REVIEW

File: sdk/core/azure-core/src/main/java/com/azure/core/http/rest/RestProxy.java [hunk 1]
```diff
@@ -545,8 +547,9 @@ private static void endTracingSpan(Signal<HttpDecodedResponse> signal) {
         // Get the context that was added to the mono, this will contain the information needed to end the span.
         reactor.util.context.Context context = signal.getContext();
         Optional<Context> tracingContext = context.getOrEmpty("TRACING_CONTEXT");
+        Optional<Boolean> disableTracing = context.getOrEmpty(Tracer.DISABLE_TRACING_KEY);
```
Comment (alzimmermsft) [inline sdk/core/azure-core/src/main/java/com/azure/core/http/rest/RestProxy.java:550]: Do we want to do the same thing as above where we use a `orElse`?

**Review by alzimmermsft — APPROVED** [review]

File: sdk/core/azure-core/src/main/java/com/azure/core/http/rest/RestProxy.java [hunk 2]
```diff
@@ -180,13 +181,14 @@ public Object invoke(Object proxy, final Method method, Object[] args) {
...
-     *
-     * @param method Service method being called.
+     * @param method  Service method being called.
      * @param context Context information about the current service call.
+     *
      * @return The updated context containing the span context.
      */
     private Context startTracingSpan(Method method, Context context) {
-        if (!TracerProxy.isTracingEnabled()) {
+        boolean disableTracing =  (boolean) context.getData(Tracer.DISABLE_TRACING_KEY).orElse(true);
```
Comment (samvaity) [inline sdk/core/azure-core/src/main/java/com/azure/core/http/rest/RestProxy.java:190]: Should this be `orElse(false)` , in the cases when a user doesn't want to disable explicitly and have it enabled by default?

Comment (srnagar) [reply sdk/core/azure-core/src/main/java/com/azure/core/http/rest/RestProxy.java]: Fixed.

File: sdk/core/azure-core/src/main/java/com/azure/core/http/rest/RestProxy.java (hunk 1 above)
Comment (srnagar) [reply sdk/core/azure-core/src/main/java/com/azure/core/http/rest/RestProxy.java]: Done.

**Pull request merged**